x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -98,167 +51,114 @@ export function ServiceOverviewTransactionsTable(props: Props) {
     sort: DEFAULT_SORT,
   });
 
+  const { pageIndex, sort } = tableOptions;
+
+  const { transactionType } = useApmServiceContext();
   const {
-    data = {
-      totalItemCount: 0,
-      items: [],
-      tableOptions: {
-        pageIndex: 0,
-        sort: DEFAULT_SORT,
-      },
-    },
-    status,
-  } = useFetcher(
+    uiFilters,
+    urlParams: { start, end, latencyAggregationType },
+  } = useUrlParams();
+
+  const { data = INITIAL_STATE, status, requestId } = useFetcher(
```

Comment:
This file is getting quite big. It might help if you extract the data fetching, including some of the pagination logic eg:

```ts
const {
    currentPagePrimaryStatistics,
    primaryStatisticsStatus,
    totalCount,
    requestId,
  } = usePrimaryStatisticsFetcher({ serviceName, pageIndex, sort });
```

You can do the same with the comparison statistics